frontend/src/components/search/SearchPagination.tsx

```diff
@@ -31,6 +33,10 @@ export default function SearchPagination({
 }: SearchPaginationProps) {
   const { pending } = useFormStatus();
 
+  const debouncedHandlePageChange = useDebouncedCallback((newPage: number) => {
+    handlePageChange(newPage);
+  }, DEBOUNCE_TIME);
```

Comment:
Basically yes - I wanted to have a quick meeting also - that's why this is still in draft.

I also would need to update some of the e2e tests to wait for page change

Reply:
Sounds good. Let's meet to talk it thru. There might be a different UX we could use — e.g. instead of debouncing, disable the pagination while results are loading.